### PR TITLE
Tools: autotest of vibration failsafe

### PR DESCRIPTION
**Pull request by rmackay9**
This autotest will hopefully ensure that the vibration failsafe remains operational.

The test does the following:

1. takes off in Loiter to 20m
2. sets the simulated accelerometer biases to 2m/s/s to simulate a high vibration environment
3. waits for the "Vibration compensation ON" message to appear
4. waits for the vehicle to switch to LAND mode (as part of the EKF failsafe)
5. waits up to 30 seconds for the vehicle to descend to 2m

This has been tested in SITL and I confirmed that it passed with master but failed if PR https://github.com/ArduPilot/ardupilot/pull/17837 was reverted.

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -1044,6 +1044,38 @@ def test_battery_failsafe(self, timeout=300):
 
         self.progress("All Battery failsafe tests complete")
 
+    # Tests the vibration failsafe
+    def test_vibration_failsafe(self, timeout=200):
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:1048]:
```suggestion
    def test_vibration_failsafe(self):
```

Unused

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -1044,6 +1044,38 @@ def test_battery_failsafe(self, timeout=300):
...
+        self.progress("Test vibration failsafe")
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:1049]: This should be redundant in the text output - we should be emitting a banner already

File: Tools/autotest/arducopter.py [hunk 3]
```diff
@@ -1044,6 +1044,38 @@ def test_battery_failsafe(self, timeout=300):
...
+        self.wait_altitude(-5, 2, timeout=30, relative=True)
+
+        # force disarm of vehicle (it will likely not automatically disarm) 
+        self.disarm_vehicle(force=True)
+
+        # revert simulated accel bias and reboot to restore EKF health
+        self.context_pop()
+        self.reboot_sitl()
+
+        self.progress("Vibration failsafe test complete")
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:1077]: This should be redundant - we should be emitting the test result.

**Review by peterbarker — APPROVED** [review]
LGTM

File: Tools/autotest/arducopter.py (hunk 1 above)
Comment (rmackay9) [reply Tools/autotest/arducopter.py]: removed, txs

File: Tools/autotest/arducopter.py (hunk 2 above)
Comment (rmackay9) [reply Tools/autotest/arducopter.py]: removed text, txs

File: Tools/autotest/arducopter.py (hunk 3 above)
Comment (rmackay9) [reply Tools/autotest/arducopter.py]: removed, thanks.

**Comment by peterbarker** [comment]
Force-pushed fix for Python flake8 test - trailing whitespace.

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**